torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -23,6 +23,45 @@ using torch::autograd::variable_list;
 static constexpr char* kNumBackwardPasses = "num_current_backward_passes";
 static constexpr char* kNumAutogradContexts = "num_autograd_contexts";
 
+// This hook does 2 things:
+//   1. Accumuate the gard to RPC context.
+//   2. Call post hooks of the original AccumulateGrad.
+struct DistAccumulateGradCapturePreHook
```

Comment:
nit: If we're renaming GradCapturePreHook as I suggested above, we can rename this too.